aesara/tensor/subtensor_opt.py

```diff
@@ -1646,23 +1646,18 @@ def local_subtensor_SpecifyShape_lift(fgraph, node):
         return False
 
     obj_arg = specify_shape_node.owner.inputs[0]
-    shape_arg = specify_shape_node.owner.inputs[1]
+    shape_arg = specify_shape_node.owner.inputs[1:]
 
     indices = get_idx_list(node.inputs, node.op.idx_list)
 
-    if len(indices) > 1 or len(indices) == 0:
-        return False
-
-    if isinstance(indices[0], slice) or isinstance(
-        getattr(indices[0], "type", None), SliceType
+    if any(
+        isinstance(index, slice) or isinstance(getattr(index, "type", None), SliceType)
+        for index in indices
```

Comment:
Just as I mentioned above, whenever we encounter a `None` we can use `aesara.tensor.shape.shape_i`.

Reply:
This require no longer relies on specifyshape input types, it doesn't matter if it's a `None` or not.